02-microarray/pathway-analysis_microarray_04_gsva.Rmd

```diff
@@ -0,0 +1,631 @@
+mapped_df %>%
+  dplyr::filter(entrez_id %in% dup_entrez_ids) %>%
+  dplyr::select(Ensembl, entrez_id, mean, dplyr::everything())
+```
+
+Now we can filter out the duplicate gene identifiers using the mean expression value.
+
+```{r}
+filtered_mapped_df <- mapped_df %>%
+  # Sort so that the highest absolute mean expression values are at the top
+  dplyr::arrange(dplyr::desc(abs(mean))) %>%
+  # Filter out the duplicated rows using `dplyr::distinct()`-- this will keep
+  # the first row with the duplicated value thus keeping the row with the
+  # highest absolute value of the mean expression
+  dplyr::distinct(entrez_id, .keep_all = TRUE)
+```
+
+Let's check to see that we removed the duplicate Entrez IDs and kept the rows with the higher absolute value of the mean expression.
+
+```{r}
+filtered_mapped_df %>%
+  dplyr::filter(entrez_id %in% dup_entrez_ids) %>%
+  dplyr::select(Ensembl, entrez_id, mean, dplyr::everything())
+```
+
+Looks like we were able to successfully get rid of the duplicate gene identifiers and keep the observations with the higher absolute mean expression value!
+
+As mentioned earlier, we need a matrix for GSVA.
+Let's now convert our data frame into a matrix and prepare our object for GSVA.
+
+```{r}
+matrix <- filtered_mapped_df %>%
+  # First let's remove the columns we no longer need
+  dplyr::select(-c("Ensembl", "mean")) %>%
+  # Now we need to store our gene identifiers as rownames
+  tibble::column_to_rownames("entrez_id") %>%
+  # Now we can convert our object into a matrix
+  as.matrix()
+```
+
+Note that if we had duplicate gene identifiers here, we would not be able to set them as rownames.
+
+## GSVA - Microarray
+
+**DRAFT**
+
+You may notice that GSVA has some commonalities with GSEA.
+Rather than ranking genes based on some statistic _we_ selected ahead of time, GSVA fits a model and ranks genes based on their expression level relative to the sample distribution.
+This is a way of asking if a gene _i_ is highly or lowly expressed in a sample _j_ in the context of this experiment and ranking accordingly ([Hänzelmann2013](https://doi.org/10.1186/1471-2105-14-7)).
+The pathway-level score calculated is a way of asking how genes _within_ a gene set vary as compared to genes that are _outside_ of that gene set ([Malhotra. 2018](https://towardsdatascience.com/decoding-gene-set-variation-analysis-8193a0cfda3)). 
+(This is sometimes called a competitive test in gene set enrichment literature.)
+The intuition here is that we will get pathway-level scores for each sample that indicate if genes in a pathway vary concordantly in one direction (over-expressed or under-expressed relative to the overall population) ([Hänzelmann2013](https://doi.org/10.1186/1471-2105-14-7)).
+
+The output is a gene set by sample matrix of GSVA scores.
+
+### Perform GSVA
+
+**REVIEW**
+
+```{r}
+gsva_results <- gsva(matrix,
+  hallmarks_list,
+  method = "gsva",
+  # Appropriate for our transformed data on log2-like scale
```

Comment:
Because this is a SCAN processed microarray dataset, you probably can say:
```suggestion
  # Appropriate for our log2-transformed microarray data
```